### PR TITLE
fix spCreateChecks subquery

### PR DESCRIPTION
**Pull request by GuyHarwood**
correction to the subquery that produces an unused pin for a particular school

### CODE REVIEW

File: admin/data/sql/migrations/153.do.fix-spCreateChecks.sql [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+      FROM @TVP
+    FOR READ ONLY
+
+  DECLARE @pupilId int
+  DECLARE @checkFormId int
+  DECLARE @checkWindowId int
+  DECLARE @isLiveCheck bit
+  DECLARE @pinExpiresAt datetimeoffset
+  DECLARE @schoolId int
+  DECLARE @pin int
```
Comment (jon-shipley) [inline admin/data/sql/migrations/153.do.fix-spCreateChecks.sql]: @pin is not used?

File: admin/data/sql/migrations/153.do.fix-spCreateChecks.sql [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  DECLARE @err int
```
Comment (jon-shipley) [inline admin/data/sql/migrations/153.do.fix-spCreateChecks.sql]: @err is not used?

File: admin/data/sql/migrations/153.do.fix-spCreateChecks.sql [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    VALUES (@pupilId, @checkFormId, @checkWindowId, @isLiveCheck);
+
+    -- Get the check.id we just inserted
+    SET @checkId = SCOPE_IDENTITY();
+
+    -- Assign a pin to the check
+    INSERT INTO [mtc_admin].[checkPin] (school_id, check_id, pinExpiresAt, pin_id)
+    VALUES (@schoolId, @checkId, @pinExpiresAt, (SELECT TOP 1 p.id FROM mtc_admin.pin p 
+                    LEFT OUTER JOIN (SELECT cp.pin_id, cp.school_id FROM mtc_admin.checkPin cp WHERE school_id = @schoolId) 
+                    AS vew ON p.id = vew.pin_id WHERE vew.pin_id IS NULL))
```
Comment (jon-shipley) [inline admin/data/sql/migrations/153.do.fix-spCreateChecks.sql]: Is it worth ordering by NEWID() on the select?

File: admin/data/sql/migrations/153.do.fix-spCreateChecks.sql [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    INSERT INTO [mtc_admin].[check]
+        (pupil_id, checkForm_id, checkWindow_id, isLiveCheck)
+    VALUES (@pupilId, @checkFormId, @checkWindowId, @isLiveCheck);
+
+    -- Get the check.id we just inserted
+    SET @checkId = SCOPE_IDENTITY();
+
+    -- Assign a pin to the check
+    INSERT INTO [mtc_admin].[checkPin] (school_id, check_id, pinExpiresAt, pin_id)
+    VALUES (@schoolId, @checkId, @pinExpiresAt, (SELECT TOP 1 p.id FROM mtc_admin.pin p
```
Comment (GeorgeChatzigiannis) [inline admin/data/sql/migrations/153.do.fix-spCreateChecks.sql]: @GuyHarwood minor comment: can you split that sql in different lines to make it more readable please?

**Pull request merged**